components/datetime/src/skeleton.rs

```diff
@@ -610,10 +605,40 @@ pub fn get_best_available_format_pattern<'a>(
     if closest_missing_fields == fields.len() {
         return BestSkeleton::NoMatch;
     }
+
+    if closest_distance == NO_DISTANCE {
+        return BestSkeleton::AllFieldsMatch(closest_format_pattern.clone());
+    }
+
+    // Modify the resulting pattern to have fields of the same length.
+    let expanded_pattern = Pattern::from(
+        closest_format_pattern
+            .items()
+            .iter()
+            .map(|item| {
+                if let PatternItem::Field(pattern_field) = item {
+                    if let Some(requested_field) = fields
+                        .iter()
+                        .find(|field| field.symbol == pattern_field.symbol)
+                    {
+                        if requested_field.length != pattern_field.length
+                            && requested_field.get_length_type() == pattern_field.get_length_type()
```

Comment:
`suggestion (n-b)`: the only uses of `get_length_type` is to retrieve it for two fields and compare if they're equal. Could we optimize it a bit by having `field.matches_length(&self, field) -> bool`?

Reply:
I'm not sure what you mean by "optimize", but I would prefer the current implementation to:

```diff
diff --git a/components/datetime/src/fields/mod.rs b/components/datetime/src/fields/mod.rs
index 30652acc..829f8fc8 100644
--- a/components/datetime/src/fields/mod.rs
+++ b/components/datetime/src/fields/mod.rs
@@ -31,7 +31,11 @@ pub struct Field {
 }
 
 impl Field {
-    pub fn get_length_type(&self) -> TextOrNumeric {
+    pub fn matches_length_type(&self, field: &Field) -> bool {
+        self.get_length_type() == field.get_length_type()
+    }
+
+    fn get_length_type(&self) -> TextOrNumeric {
         match self.symbol {
             FieldSymbol::Year(year) => year.get_length_type(self.length),
             FieldSymbol::Month(month) => month.get_length_type(self.length),
diff --git a/components/datetime/src/skeleton.rs b/components/datetime/src/skeleton.rs
index 7f5836a6..dd4f358a 100644
--- a/components/datetime/src/skeleton.rs
+++ b/components/datetime/src/skeleton.rs
@@ -565,11 +565,10 @@ pub fn get_best_available_format_pattern(
                         NO_DISTANCE
                     } else if requested_field.symbol != skeleton_field.symbol {
                         SUBSTANTIAL_DIFFERENCES_DISTANCE
-                    } else if requested_field.get_length_type() != skeleton_field.get_length_type()
-                    {
-                        TEXT_VS_NUMERIC_DISTANCE
-                    } else {
+                    } else if requested_field.matches_length_type(&skeleton_field) {
                         WIDTH_MISMATCH_DISTANCE
+                    } else {
+                        TEXT_VS_NUMERIC_DISTANCE
                     };
 
                     requested_fields.next();
@@ -622,7 +621,7 @@ pub fn get_best_available_format_pattern(
                         .find(|field| field.symbol == pattern_field.symbol)
                     {
                         if requested_field.length != pattern_field.length
-                            && requested_field.get_length_type() == pattern_field.get_length_type()
+                            && requested_field.matches_length_type(&pattern_field)
                         {
                             return PatternItem::Field(*requested_field);
                         }
```

Reply:
yup, that's what I suggesed. If you prefer the current API, I'm okay with that!